api/notd/block_processor.py

```diff
@@ -59,12 +77,77 @@ async def get_transfers_in_block(self, blockNumber: int) -> List[RetrievedTokenT
         blockHash = block['hash'].hex()
         blockDate = datetime.datetime.fromtimestamp(block['timestamp'])
-        events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc721TansferEventSignatureHash])
-        logging.info(f'Found {len(events)} events in block #{blockNumber}')
-        allTokenTransfers = []
-        for eventsChunk in list_util.generate_chunks(events, 5):
-            allTokenTransfers += await asyncio.gather(*[self._process_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in eventsChunk])
-        return [tokenTransfer for tokenTransfer in allTokenTransfers if tokenTransfer]
+        erc721TokenTransfers = []
+        erc721events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc721TansferEventSignatureHash])
+        logging.info(f'Found {len(erc721events)} events in block #{blockNumber}')
+        for erc721EventsChunk in list_util.generate_chunks(erc721events, 5):
+            erc721TokenTransfers += await asyncio.gather(*[self._process_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc721EventsChunk])
+
+        erc1155TokenTransfers = []
+        erc1155events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferEventSignatureHash])
+        logging.info(f'Found {len(erc1155events)} erc1155events in block #{blockNumber}')
+        for erc1155EventsChunk in list_util.generate_chunks(erc1155events, 5):
+            erc1155TokenTransfers += await asyncio.gather(*[self._process_erc1155_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155EventsChunk])
+
+        erc1155BatchTokenTransfers = []
+        erc1155Batchevents = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferBatchEventSignatureHash])
+        logging.info(f'Found {len(erc1155Batchevents)} erc1155Batchevents in block #{blockNumber}')
+        for erc1155BatchEventsChunk in list_util.generate_chunks(erc1155Batchevents, 5):
+            erc1155BatchTokenTransfers += await asyncio.gather(*[self._process_erc1155_batch_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155BatchEventsChunk])
+
+        tokenTransferList = [tokenTransfer for tokenTransfer in erc1155BatchTokenTransfers for tokenTransfer in tokenTransfer] + [tokenTransfer for tokenTransfer in erc1155TokenTransfers if tokenTransfer] + [tokenTransfer for tokenTransfer in erc721TokenTransfers if tokenTransfer]
+        return  tokenTransferList
+
+    async def _process_erc1155_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[RetrievedTokenTransfer]:
```

Comment:
make this return type a ist too and replace return None with return [] and change the last line to return [transaction]

Reply:
do the same with _process_event, and please rename that to _process_erc721_event

Reply:
I thought only batch returns a list.